neovim/ui/gtk_ui.py

```diff
@@ -152,7 +152,11 @@ def _nvim_resize(self, columns, rows):
         self._cell_pixel_width = cell_pixel_width
         self._cell_pixel_height = cell_pixel_height
         self._screen = Screen(columns, rows)
-        self._window.resize(pixel_width, pixel_height)
+        prev_rect = self._drawing_area.get_allocation()
```

Comment:
Sounds like a good plan. I thought there was a method to set the preferred the size of a GtkDrawingArea (but not as as strict as `gtk_widget_set_size_request` that permanelty sets the minimum allowed size) but I can't find it again.
If we were to develop this gui futher, I think we eventually should  split the gui out from the python client/host repo, as one might want to fork/experiment with the gui, but still follow the upstream python-client code. Ping @tarruda